Storage/Storage/Site.swift

```diff
@@ -4,12 +4,81 @@
 
 import UIKit
 
-public class Site {
+protocol Identifiable {
+    var id: Int? { get set }
+}
+
+public enum IconType: Int {
+    case Icon = 0
+    case AppleIcon = 1
+    case AppleIconPrecomposed = 2
+    case Guess = 3
+}
+
+public class Favicon : Identifiable {
+    var id: Int? = nil
+    var img: UIImage? = nil
+    var filename: String {
+        if let url = NSURL(string: self.url) {
+            if let ext = url.pathExtension {
+                return "\(url.hash).\(ext)"
+            }
+        }
+        return "\(url.hash)"
+    }
+
+    public let url: String
+    public let date: NSDate
+    public let width: Int?
+    public let height: Int?
+    public let type: IconType
+
+    public func getImage(files: FileAccessor) -> UIImage? {
```

Comment:
It takes an argument (the files accessor). Storing a reference to it in every favicon would break our current setup here.